test/e2e/framework/pod/pod_client.go

```diff
@@ -192,6 +221,19 @@ func (c *PodClient) DeleteSync(ctx context.Context, name string, options metav1.
 	framework.ExpectNoError(WaitForPodNotFoundInNamespace(ctx, c.f.ClientSet, name, c.namespace, timeout), "wait for pod %q to disappear", name)
 }
 
+// addTestOrigin adds annotations to help identifying tests which incorrectly leak pods because insufficient cleanup
+func (c *PodClient) setOwnerAnnotation(pod *v1.Pod) {
+	if !GlobalTrackOwner && !c.trackOwner {
```

Comment:
With this check, it's impossible to set `GlobalTrackOwner=true` and then disable it in the `PodClient` instance.

Constructing the `PodClient` must initialize itself with the `GlobalTrackOwner`, then only `c.trackOwner` gets checked here.

Reply:
right. Done, thanks